### ISSUE TITLE
Modulo operator doesn't cache result

### ISSUE THREAD

**Issue by eventualbuddha** [juliankrispel/decaf#6]
```coffee
a %% b
```

Throws an error:

```
Error: "%%" does not match field "operator": == | != | === | !== | < | <= | > | >= | << | >> | >>> | + | - | * | / | % | & | | | ^ | in | instanceof | .. of type BinaryExpression
  at add (/Users/donovan/lib/decaf/node_modules/ast-types/lib/types.js:576:27)
  at /Users/donovan/lib/decaf/node_modules/ast-types/lib/types.js:589:17
```

**Comment by juliankrispel** [comment]
@eventualbuddha awesome thanks for all the issues :)

**Comment by juliankrispel** [comment]
Argh this is not quite done, as @eventualbuddha noted:

> This needs more work. For example, a() %% b() turns into (a() % b() + b()) % b(); which evaluates the values too many times.

**Comment by juliankrispel** [comment]
fixed in #93
